File: Backend/retail_management_system/src/main/java/com/rms/service/WholesalerService.java
package com.rms.service;

import com.rms.dto.UpdateWholesalerDTO;
import com.rms.dto.WholesalerDTO;

public interface WholesalerService {
    WholesalerDTO getProfile(Long userId);
    WholesalerDTO updateProfile(Long userId, UpdateWholesalerDTO updateDTO);
}

File: Frontend/src/services/wholesalerService.ts
import { api } from "./api";
import type { WholesalerProfile, UpdateWholesalerDTO } from "../types/wholesaler";

export const wholesalerService = {
  async getProfile(): Promise<WholesalerProfile> {
    return api.request<WholesalerProfile>("/wholesaler/profile", {
      method: "GET",
    });
  },

  async updateProfile(data: UpdateWholesalerDTO): Promise<WholesalerProfile> {
    return api.request<WholesalerProfile>("/wholesaler/profile", {
      method: "PUT",
      data,
    });
  },
};